apm-agent-core/src/main/java/co/elastic/apm/agent/logging/LoggingConfiguration.java

```diff
@@ -169,11 +171,15 @@ static String getActualLogFile(@Nullable String agentHome, String logFile) {
         if (!logDir.exists()) {
             logDir.mkdir();
         }
-        if (!logDir.canWrite()) {
+        if (!Files.isWritable(Paths.get(logFile)) && !logDir.canWrite()) {
             System.err.println("Log file " + logFile + " is not writable. Falling back to System.out.");
             return SYSTEM_OUT;
         }
-        System.out.println("Writing Elastic APM logs to " + logFile);
+
+        if(!logFile.equalsIgnoreCase("/dev/null")) {
+            System.out.println("Writing Elastic APM logs to " + logFile);
+        }
+
```

Comment:
Although this isn't very elegant, it seems to be the only way to retain the actual behavior while making the agent completely silent in case of `/dev/null`.